gax/src/main/java/com/google/api/gax/tracing/SpanName.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.tracing;
+
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalExtensionOnly;
+import com.google.auto.value.AutoValue;
+
+/** A value class to represent the name of the operation in an {@link ApiTracer}. */
+@BetaApi("Surface for tracing is not yet stable")
+@InternalExtensionOnly
+@AutoValue
+public abstract class SpanName {
+  /**
+   * Creates a new instance of the name.
+   *
+   * @param clientName The name of the client. ie BigtableData
+   * @param methodName The name of the logical operation being traced. ie. ReadRows.
+   */
+  public static SpanName of(String clientName, String methodName) {
+    return new AutoValue_SpanName(clientName, methodName);
+  }
+
+  /** The name of the client. ie BigtableData */
+  public abstract String getClientName();
+
+  /** The name of the logical operation being traced. ie. ReadRows. */
+  public abstract String getMethodName();
+
+  /** Creates a new instance with the clientName overriden. */
+  public SpanName withClientName(String clientName) {
```

Comment:
Why would we need to override client name? It seems like an impossible scenario. If there is no real need to have it, please remove the method (or clarify what exactly "client" means here)

Reply:
This is needed for manual wrappers like bigtable. I need to disambiguate between traces created by the gapic layer and the manually written callable chain. The outer bigtable traces will be called something like `BigtableData.ReadRows` while the underlying gapic generated child traces will be named `BaseBigtableData.ReadRows`

Reply:
But in that case is it still "client". Is'nt it "caller", "layer" or something? My understanding is that there is one client: bigtable client. This is what a user expects: he has a client (bigtable) and that is it. I'm find with keeping this as long as it is clear what "client" is here (it can be done without serivce-specific names, just refering to them as "GAPIC (generated) client" and "manual wrapper client around the generated one".

Reply:
But in that case is it still "client". Is'nt it "caller", "layer" or something? My understanding is that there is one client: bigtable client. This is what a user expects: he has a client (bigtable) and that is it. I'm fine with keeping this as long as it is clear what "client" is here (it can be done without serivce-specific names, just refering to them as "GAPIC (generated) client" and "manual wrapper client around the generated one".